fs/walk.go

```diff
@@ -250,6 +250,15 @@ func (dt DirTree) Dirs() (dirNames []string) {
 	return dirNames
 }
 
+// Prune remove directories from a directory tree.
+func (dt DirTree) Prune(dirNames []string) {
+	// We do not need to do a recursion.
+	// GC will take care of deleting child entries.
+	for _, dName := range dirNames {
+		delete(dt, dName)
```

Comment:
`delete`-ing the entry is problematic as there will be a DirEntry of the deleted directory in the parent.

This leads to this with `rclone tree`

```
/
├── a
├── /sub [Couldn't find directory "sub"]
├── sub1
│   └── file1
└── sub2
    └── file2
```

I think setting the entry to nil (ie an empty slice) will work better

```
dt[dName] = nil
```